lib/schedule/minischedule/load.ex

```diff
@@ -37,13 +37,37 @@ defmodule Schedule.Minischedule.Load do
     }
   end
 
-  @spec run(Run.key(), [Activity.t()] | nil, [Trip.t()] | nil, %{Block.key() => [Trip.t()]}) ::
+  @spec run(
+          Run.key(),
+          [Activity.t()] | nil,
+          [Trip.t()] | nil,
+          %{Block.key() => [Trip.t()]},
+          Schedule.Trip.by_id()
+        ) ::
           Run.t()
-  def run(run_key, activities, trips, all_trips_by_block) do
+  def run(run_key, activities, trips, all_trips_by_block, trips_by_id) do
     {schedule_id, run_id} = run_key
     activities = activities || []
     trips = trips || []
 
+    service_ids =
+      trips
+      |> Enum.map(& &1.trip_id)
+      |> Enum.map(&Map.get(trips_by_id, &1))
+      |> Enum.filter(&(!is_nil(&1)))
```

Comment:
At least IMHO it's desirable to avoid `!` for negation when there's a reasonable alternative: I find it tends to get swallowed up visually, leading to confusion. Here and just below we could `Enum.reject(& is_nil(&1))`